src/Sulu/Bundle/WebsiteBundle/Analytics/AnalyticsManager.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\WebsiteBundle\Analytics;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Bundle\WebsiteBundle\Entity\Analytic;
+use Sulu\Bundle\WebsiteBundle\Entity\AnalyticRepository;
+use Sulu\Bundle\WebsiteBundle\Entity\Domain;
+use Sulu\Bundle\WebsiteBundle\Entity\DomainRepository;
+
+/**
+ * Manages analytics.
+ */
+class AnalyticsManager implements AnalyticsManagerInterface
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var AnalyticRepository
+     */
+    private $repository;
+
+    /**
+     * @var DomainRepository
+     */
+    private $domainRepository;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        AnalyticRepository $repository,
+        DomainRepository $domainRepository
+    ) {
+        $this->entityManager = $entityManager;
+        $this->repository = $repository;
+        $this->domainRepository = $domainRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findAll($webspaceKey)
+    {
+        return $this->repository->findByWebspaceKey($webspaceKey);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function find($id)
+    {
+        return $this->repository->findById($id);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($webspaceKey, $data)
+    {
+        $entity = new Analytic();
+        $this->setData($entity, $webspaceKey, $data);
+
+        $this->entityManager->persist($entity);
+        $this->entityManager->flush();
+
+        return $entity;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($id, $data)
+    {
+        $entity = $this->find($id);
+        $this->setData($entity, $entity->getWebspaceKey(), $data);
+
+        $this->entityManager->flush();
+
+        return $entity;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove($id)
+    {
+        $this->entityManager->remove($this->entityManager->getReference(Analytic::class, $id));
+        $this->entityManager->flush();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function removeMultiple($ids)
+    {
+        foreach ($ids as $id) {
+            $this->entityManager->remove($this->entityManager->getReference(Analytic::class, $id));
+        }
+
+        $this->entityManager->flush();
```

Comment:
Didn't we say that we should flush at the very end (meaning the controller or the command)? Then this method would probably be not necessary. /cc @chirimoya